<?xml version="1.0" ?>
<!--Taken and modified from https://github.com/dan-riley/octomap_merger-->
<launch>
  <arg name="vehicle" default="Megabot" />
  <arg name="ns" default="/$(arg vehicle)" />
  <!-- Type of node.  Current options are 'robot' and 'base' -->
  <arg name="type" default="robot" />
  <!-- Whether to merge everything (true) or prioritize own map (false) -->
  <arg name="fullMerge" default="true" />
  <!-- Whether to prioritize free space from any agent -->
  <arg name="freePrioritize" default="false" />
  <arg name="publishMergedPCL" default="true" />
  <arg name="publishDiffPCL" default="false" />
  <!-- Map resolution.  If different res needed for each need to change code
  	MAY need to adjust to larger values for performance. -->
  <arg name="resolution" default="0.01" />
  <!-- Rate to run the node at. Entirely bottlenecked by resources available...
	i.e. after a certain point, no matter how fast this is, the publish rate 
	WILL NOT increase. -->
  <arg name="rate" default="20.0" />
  <!-- Size of map differences to trigger a merge... if NOT full merging. -->
  <arg name="mapThresh" default="50" />

  <!-- Binary Map -->
  <!-- Topics to subscribe and publish -->
  <arg name="mapTopic" default="octomap_binary1" />
  <arg name="neighborsTopic" default="binary_neighbors_topic1" />
  <arg name="mergedTopic" default="binary_merged_octomap1" />
  <arg name="mapDiffsTopic" default="binary_octomap_diffs1" />
  <arg name="numDiffsTopic" default="binary_octonum_diffs1" />
  <arg name="pclMergedTopic" default="binary_octomap_pc2_merged1" />
  <arg name="pclDiffTopic" default="binary_octo_map_pc2_diff1" />
  <!-- Octomap type - 0: Binary, 1: Full -->
  <arg name="octoType" default="0" />

  <node name="asus_xtion_pro_binary_octomap_merger" pkg="octomap_merger" type="octomap_merger_node" output="screen">
    <param name="vehicle" value="$(arg vehicle)" />
    <param name="type" value="$(arg type)" />
    <param name="fullMerge" value="$(arg fullMerge)" />
    <param name="publishMergedPCL" value="$(arg publishMergedPCL)" />
    <param name="publishDiffPCL" value="$(arg publishDiffPCL)" />
    <param name="octoType" value="$(arg octoType)" />
    <param name="resolution" value="$(arg resolution)" />
    <param name="rate" value="$(arg rate)" />
    <param name="mapThresh" value="$(arg mapThresh)" />
    <param name="mapTopic" value="$(arg mapTopic)" />
    <param name="neighborsTopic" value="$(arg neighborsTopic)" />
    <param name="mergedTopic" value="$(arg mergedTopic)" />
    <param name="mapDiffsTopic" value="$(arg mapDiffsTopic)" />
    <param name="numDiffsTopic" value="$(arg numDiffsTopic)" />
    <param name="pclMergedTopic" value="$(arg pclMergedTopic)" />
    <param name="pclDiffTopic" value="$(arg pclDiffTopic)" />
  </node>

</launch>
